APMrover2/mode_rtl.cpp

```diff
@@ -9,37 +9,47 @@ bool ModeRTL::_enter()
     }
 
     // initialise waypoint speed
-    set_desired_speed_to_default(true);
+    if (is_positive(g2.rtl_speed)) {
+        g2.wp_nav.set_desired_speed(g2.rtl_speed);
+    } else {
+        g2.wp_nav.set_desired_speed_to_default();
+    }
 
     // set target to the closest rally point or home
 #if AP_RALLY == ENABLED
-    set_desired_location(rover.g2.rally.calc_best_rally_or_home_location(rover.current_loc, ahrs.get_home().alt));
+    g2.wp_nav.set_desired_location(g2.rally.calc_best_rally_or_home_location(rover.current_loc, ahrs.get_home().alt));
 #else
     // set destination
-    set_desired_location(rover.home);
+    g2.wp_nav.set_desired_location(rover.home);
 #endif
 
+    sent_notification = false;
+
     return true;
 }
 
 void ModeRTL::update()
 {
-    // calculate distance to home
-    _distance_to_destination = rover.current_loc.get_distance(_destination);
-    const bool near_wp = _distance_to_destination <= rover.g.waypoint_radius;
-    // check if we've reached the destination
-    if (!_reached_destination && (near_wp || rover.current_loc.past_interval_finish_line(_origin, _destination))) {
-        // trigger reached
-        _reached_destination = true;
-        gcs().send_text(MAV_SEVERITY_INFO, "Reached destination");
-    }
     // determine if we should keep navigating
-    if (!_reached_destination || (rover.is_boat() && !near_wp)) {
-        // continue driving towards destination
-        calc_steering_to_waypoint(_reached_destination ? rover.current_loc :_origin, _destination, _reversed);
-        calc_throttle(calc_reduced_speed_for_turn_or_distance(_reversed ? -_desired_speed : _desired_speed), true, true);
+    if (!g2.wp_nav.reached_destination() || rover.is_boat()) {
```

Comment:
This will mean boats never send notification that they reached destination. I think we would want to put this check in the else below:

```
if (rover.is_boat() {
  navigate_to_waypoint();
} else {
  stop_vehicle();
}
```

It's the same that is done in SmartRTL and similar to what is done in Auto and Guided, except `start_loiter` is used there.

Reply:
Txs for this.  OK, I'm changing it so that it attempts to Loiter instead of continuing to navigate.  This is a change in behaviour that I was going to avoid but it's an improvement and fits with [this issue](https://github.com/ArduPilot/ardupilot/issues/11285).